mlflow/llama_index/__init__.py

```diff
@@ -93,13 +94,22 @@ def _get_llama_index_version() -> str:
         )
 
 
+def _supported_classes():
+    from llama_index.core.base.base_query_engine import BaseQueryEngine
+    from llama_index.core.chat_engine.types import BaseChatEngine
+    from llama_index.core.indices.base import BaseIndex
+    from llama_index.core.retrievers import BaseRetriever
+
+    return BaseIndex, BaseChatEngine, BaseQueryEngine, BaseRetriever
+
+
 @experimental
 @format_docstring(LOG_MODEL_PARAM_DOCS.format(package_name=FLAVOR_NAME))
 @trace_disabled  # Suppress traces while loading model
 def save_model(
-    index,
+    llama_model,
```

Comment:
Changing the argument name to `llama_model` as it is no longer limited to an index. This is a breaking change, but I considered it is ok because it only blocks logging a new model and requires very small migration effort. But we may want to do more strict deprecation process if it's necessary: cc @BenWilson2

Reply:
Given llama meta's model, I think we should rename to `llama_index_model` or something less confusing.

Reply:
Yea my concern was `llama_index_model` is a bit long so tried to make it as short as others like `lc_model`. But now I found `transformers_model` exists, then we don't likely have any issue with the full name🫠